service/header_service_test.go

```diff
@@ -0,0 +1,288 @@
+			expectedCount: 0,
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range testCases {
+		headers, err := tData.hs.Headers.GetHeaderAncestorsByHash(tt.hash, tt.ancestorHash)
+
+		assert.Equal(t, err != nil, tt.expectedError)
+		assert.Equal(t, len(headers), tt.expectedCount)
+	}
+}
+
+func TestGetCommonAncestor(t *testing.T) {
+	tData := setUpServices()
+
+	testCases := []struct {
+		hashes         []string
+		ancestorHeight int32
+		expectedError  bool
+	}{
+		{
+			hashes: []string{
+				"000000004ebadb55ee9096c9a2f8880e09da59c0d68b1c228da88e48844a1485", // Height = 4
+				"000000006a625f06636b8bb6ac7b960a8d03705d1ace08b1a19da3fdcc99ddbd", // Height = 2
+			},
+			ancestorHeight: 1,
+			expectedError:  false,
+		},
+		{
+			hashes: []string{
+				"000000004ebadb55ee9096c9a2f8880e09da59c0d68b1c228da88e48844a1485", // Height = 4
+				"0000000082b5015589a3fdf2d4baff403e6f0be035a5d9742c1cae6295464449", // Height = 4 Fork
+				"000000006a625f06636b8bb6ac7b960a8d03705d1ace08b1a19da3fdcc99ddbd", // Height = 2
+				"00000000839a8e6886ab5951d76f411475428afc90947ee320161bbf18eb6048", // Height = 1
+			},
+			ancestorHeight: 0,
+			expectedError:  false,
+		},
+		{
+			hashes: []string{
+				"000000004ebadb55ee9096c9a2f8880e09da59c0d68b1c228da88e48844a1485", // Height = 4
+				"000000006a625f06636b8bb6ac7b960a8d03705d1ace08b1a19da3fdcc99ddbd", // Height = 2
+				"00000000839a8e6886ab5951d76f411475428afc90947ee320161bbf18eb6048", // Height = 1
+				"000000007bc154e0fa7ea32218a72fe2c1bb9f86cf8c9ebf9a715ed27fdb229a", // Height = 100
+			},
+			ancestorHeight: 0,
+			expectedError:  true,
+		},
+	}
+
+	for _, tt := range testCases {
+		header, err := tData.hs.Headers.GetCommonAncestors(tt.hashes)
+
+		assert.Equal(t, err != nil, tt.expectedError)
+		if header != nil {
+			assert.Equal(t, header.Height, tt.ancestorHeight)
+		}
+	}
+}
+
+func TestLatestHeaderLocator(t *testing.T) {
+	tData := setUpServices()
+
+	locator := tData.hs.Headers.LatestHeaderLocator()
+	headers, _ := tData.hs.Headers.GetHeadersByHeight(0, 5)
+
+	for _, header := range headers {
+		check := false
+		for _, hash := range locator {
+			if *hash == header.Hash {
+				check = true
+			}
+		}
+		assert.Equal(t, check, true)
+	}
+}
+
+func TestGetAllTips(t *testing.T) {
+	tData := setUpServices()
+
+	//Check tips without fork
+	tips, _ := tData.hs.Headers.GetTips()
+	assert.Equal(t, len(tips), 1)
+
+	//Check tip with fork
+	tData.hs.Headers.AddHeader(createFork(), blocksToConfirmFork)
+	tips, _ = tData.hs.Headers.GetTips()
+	assert.Equal(t, len(tips), 2)
+}
+
+func setUpServices() *testData {
+	db, _ := testrepository.LongestChain()
+	var array []domains.BlockHeader = db
+	repo := &repository.Repositories{
+		Headers: testrepository.NewHeadersTestRepository(&array),
+	}
+
+	hs := NewServices(Dept{
+		Repositories: repo,
+		Peers:        nil,
+	})
+
+	return &testData{
+		db: &array,
+		hs: hs,
+	}
+}
+
+func createFifthHeader() domains.BlockHeader {
```

Comment:
Maybe let's merge those two funcs as they are identical, and pass to them correct parameters creating fork and fifth header and then name the variables as:
fork,
fifthHeader